### ISSUE TITLE
Reading MATH in label strings

### ISSUE THREAD

**Issue by tlhamrick** [nvaccess/nvda#5862]
I have noticed that when a `math` element appears as part of a label string for an input, that it is being completely ignored.

When reading the math inline everything seems to work fine, but specifically when I am focused on the input, the computed label value it not able to include the text equivalent of the MathML (it will read all text elements and read nothing for the MathML elements). I have had no issues with using other types of markup text in these strings (e.g. `span`, `div`, `sup`, `sub`).

Note that we are using MathJax to render the MathML on the page if that makes any difference.

You can see what I mean if you go to look at the following jsfiddle: https://jsfiddle.net/L3zw7rtb/1/

Let me know if I can provide any further information.

**Comment by derekriemer** [comment]
You need to provide the for attribute of the label to the id of the edit 
field.

   <label for = "radio-2">
     <input type='radio' id='radio-2' name='radio-group'/>

On 4/6/2016 12:47 PM, tlhamrick wrote:
>
> I have noticed that when a |math| element appears as part of a label 
> string for an input, that it is being completely ignored.
>
> When reading the math inline everything seems to work fine, but 
> specifically when I am focused on the input, the computed label value 
> it not able to include the text equivalent of the MathML (it will read 
> all text elements and read nothing for the MathML elements). I have 
> had no issues with using other types of markup text in these strings 
> (e.g. |span|, |div|, |sup|, |sub|).
>
> Note that we are using MathJax to render the MathML on the page if 
> that makes any difference.
>
> You can see what I mean if you go to look at the following jsfiddle: 
> https://jsfiddle.net/L3zw7rtb/1/
>
> Let me know if I can provide any further information.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/5862>
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

**Comment by tlhamrick** [comment]
Well I have noticed that it is pretty common practice with radio buttons that if you wrap them inside of a <label> element that the label will still be read (as can be seen in the example that only has plain text in the label string).

But even if I add the `for` attribute to the radio buttons, I am not getting any of the math to be read, specifically when I am focused on the input itself.

See the updated example: https://jsfiddle.net/L3zw7rtb/2/

**Comment by jcsteh** [comment]
The problem here is that labels (as defined for accessibility) are only textual. There are some good reasons for this: hearing all of the semantic info for a label (e.g. the fact that it's a heading, its formatting, etc.) usually isn't useful. However, the downside is that complex content such as math can't be read as a label. Unfortunately, this isn't something we can implement in NVDA without several huge refactors. Even if we did, I'm concerned it would cause performance degradation *everywhere*, since we now have to crawl the label elements for every control rather than just fetching the control's label text.

Depends on #4877.

**Comment by pbblair** [comment]
I know that a "+1" comment doesn't help very much, but I would like to support this suggestion.

It's actually a pretty realistic use case that you would want to have math in a multiple-choice exam.  As it stands, the only option for MathML accessibility when using simple input labels is to depend on alt-text for the answer choices.  It would be great if MathML in this context could at least be spoken properly.  MathPlayer's rich interaction and Nemeth output would certainly be nice-to-have, too, but if that adds to the complexity that could be something to consider down the line.

I would imagine that #7485 would probably share a solution; am I right?  In both cases it's a matter of NVDA passing the MathML to MathPlayer when it currently isn't.

**Comment by Adriani90** [comment]
cc: @jcsteh

**Comment by seanbudd** [comment]
This is blocked by the changes to the spec
